pre-commit.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# direct output to stderr
+exec 1>&2
+
+files_changed=$(git diff --cached --name-status | awk '$1 != "D" { print $2 }' | grep \.js$)
+
+echo "pre-comit > eslint"
+[ ! -z "$files_changed" ] && node_modules/.bin/eslint $files_changed || echo "skipping"
```

Comment:
`! -z` is a double-negative. It would be more direct to say `[ -n "$files_changed" ]` or even just `[ "$files_changed" ]`, or if you use `/bin/bash` in the she-bang line, then even simpler: `[[ $files_changed ]]` ... (no quotes needed when using double brackets)

Reply:
Hmm, I don't have an `eslint` program in `node_modules/bin`. Is there a package needed in package.json for this?

Reply:
Never mind. I missed the `.` in `.bin`. `eslint` is indeed there